TrkDiag/src/TrackAnalysisReco_module.cc

```diff
@@ -243,7 +253,8 @@ namespace mu2e {
     art::EDAnalyzer(conf),
     _conf(conf()),
     _trigbitsh(0),
-    _infoMCStructHelper(conf().infoMCStructHelper())
+    _infoMCStructHelper(conf().infoMCStructHelper()),
+    _toff(conf().timeOffsets())
```

Comment:
We already set the time maps in the InfoMCStructHelper so I'd rather not have them defined twice. It looks like access to "_toff" is needed though. I would just add an accessor to InfoMCStructHelper and get the time maps from there.

Reply:
Hi Andy,

Thank you for the comment. It's a reasonable request.
The issue I'm struggling with is that the time maps in the InfoMCStructHelper point to compressed objects:
TimeMaps : [ @sequence::RecoCompression.TimeMaps ]

I'd like to use uncompressed CRV data products to study the sources of the inefficiencies.
In this case, I have to provide uncompressed time maps.
Do you have a recommendation on how to 'properly' implement this?

   Best, Yuri.

Reply:
Hi Yuri,

You should be able to give the uncompressed time maps to InfoMCStructHelper and the rest of trkana should work. Let me know if that's not the case and I will rethink.

For getting the time maps from InfoMCStructHelper to pass to `FillCrvPulseInfoCollections`. I would add a function like this:

`const SimParticleTimeOffset& getTimeMaps() const { return _toff; }`

to InfoMCStructHelper.hh and then use `_infoMCStructHelper.getTimeMaps()` in the `FillCrvPulseInfoCollections` call.

Cheers,
Andy

Reply:
Hi Andy,

I've tried exactly that. It throws an exception:
        ---- BADINPUTS BEGIN
          SimParticleTimeOffset::totalTimeOffset(): the primary (2741432267, 0) is not in an input map
I'll take another look

Reply:
For anyone interested, we resolved this by reconfiguring Yuri's reconstruction job to not compress any CrvDigiMCs for his study.